mcp-lambda-server/src/validation/engine.py

```
@@ -0,0 +1,389 @@
+    def _suggest_setter_fix(self, violation: Violation) -> FixSuggestion:
+        """Setter 메서드 수정 제안"""
+        method_name = violation.code_snippet or "setXxx"
+
+        return FixSuggestion(
+            violation=violation,
+            suggestion=f"""Setter 제거 후 불변 패턴 적용:
+1. 필드를 private final로 변경
+2. 생성자에서 초기화
+3. 상태 변경이 필요하면 새 객체 반환: {method_name.replace('set', 'with')}() 메서드""",
+            before_code=f"public void {method_name}(...) {{ ... }}",
+            after_code=f"// Setter 제거, 불변 객체로 변경",
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)